src/main/java/view_user_public_profile_usecase/ViewUserPublicProfileInteractor.java

```diff
@@ -0,0 +1,43 @@
+package view_user_public_profile_usecase;
+import Entities.User;
+
+/**
+ * The view user public profile use case.
+ */
+public class ViewUserPublicProfileInteractor implements ViewUserPublicProfileInputBoundary {
+    final private ViewUserPublicProfileDSGateway userDSGateway;
+    final private ViewUserPublicProfileOutputBoundary viewUserProfileOutputBoundary;
+
+    /**
+     *
+     * @param userDSGateway is the data access interface used to save and find the user's public profile.
+     * @param viewUserProfileOutputBoundary is the output boundary implemented by the viewUserPublicProfilePresenter
+     */
+    ViewUserPublicProfileInteractor(ViewUserPublicProfileDSGateway userDSGateway,
+                                    ViewUserPublicProfileOutputBoundary viewUserProfileOutputBoundary) {
+        this.userDSGateway = userDSGateway;
+        this.viewUserProfileOutputBoundary = viewUserProfileOutputBoundary;
+    }
+
+    /**
+     * Implements the showUserProfile from the Input Boundary's interface.
+     * @param requestModel the request model for the view public profile use case. Holds the username.
+     * @return the user's public profile information.
+     */
+    @Override
+    public ViewUserPublicProfileResponseModel showUserProfile(ViewUserPublicProfileRequestModel requestModel) {
+        /*Find User*/
+        String username = requestModel.getUsername();
+
+        User user = userDSGateway.findUser(username);
+
+        ViewUserPublicProfileResponseModel successViewResponseModel = new ViewUserPublicProfileResponseModel(
+                user.getUserPublicProfile().getPreferences(),
+                user.getUserPublicProfile().getCoursePreferences(),
+                user.getUserPublicProfile().getBiography());
+
+        /*Return response model with userPublicProfile information*/
+        return viewUserProfileOutputBoundary.prepareSuccessView(successViewResponseModel);
+    }
+}
```

Comment:
Is there a way to fail here? I cannot think of any checks (other than to check if the user exists which should not be necessary) If not, is prepare fail view required?

Reply:
Okay yeah I don't know. I was asking Sharon about this today but hopefully we'll have more clarification today when we discuss.